pkg/kubelet/pleg/generic.go

```diff
@@ -431,6 +433,24 @@ func getContainerState(pod *kubecontainer.Pod, cid *kubecontainer.ContainerID) p
 	return state
 }
 
+func updateRunningPodAndContainerMetrics(pods []*kubecontainer.Pod) {
+	// Set the number of running pods in the parameter
+	metrics.RunningPodCount.Set(float64(len(pods)))
+
+	// get number of running containers given a Pod
+	numberRunningContainers := 0
+	for _, pod := range pods {
+		containers := pod.Containers
+		for _, container := range containers {
+			// check if the state of the container is "running"
+			if container.State == kubecontainer.ContainerStateRunning {
```

Comment:
This logic is a bit different than the existing logic for the metric (which does a simpler `len(p.Containers)`). @dashpole, for more input.

Reply:
Right, I did this change because it aligns more with the logic of metric name, i.e counting the containers in the running state.

Reply:
Yeah, I agree it makes more sense; regardless, it is a change in the semantics of the existing metric so we should check with the node team to ensure it isn't likely to break anything.

Reply:
I don't think we can change it.  Its hard to reason about what can break.  Could we add the container state as a label instead?  That wouldn't change queries over the base metric, and allow doing the filtering at a later point.

Reply:
That makes total sense. 
I have now added the container state as a label `container_state` in the `running_container_count` metrics.